File: Chapter04/python/1.6.2/com/sparksamples/movie_data.py
from util import get_movie_data
import matplotlib.pyplot as plt
import matplotlib

def main():
    movie_data = get_movie_data()
    print movie_data.first()
    num_movies = movie_data.count()
    print "Movies: %d" % num_movies
    movie_fields = movie_data.map(lambda lines: lines.split("|"))
    years = movie_fields.map(lambda fields: fields[2]).map(lambda x: convert_year(x))
    # we filter out any 'bad' data points here
    years_filtered = years.filter(lambda x: x != 1900)
    # plot the movie ages histogram
    movie_ages = years_filtered.map(lambda yr: 1998-yr).countByValue()
    values = movie_ages.values()
    bins = movie_ages.keys()
    plt.hist(values, bins=bins, color='lightblue', normed=True)
    fig = matplotlib.pyplot.gcf()
    fig.set_size_inches(16,10)
    plt.show()


def convert_year(x):
    try:
        return int(x[-4:])
    except:
        # there is a 'bad' data point with a blank year, which we set to 1900 and will filter out later
        return 1900


if __name__ == "__main__":
    main()

File: Chapter04/scala/1.6.2/src/main/scala/org/sparksamples/MovieData.scala
package org.sparksamples

/**
  * Created by Rajdeep on 12/22/15.
  */
import scala.collection.immutable.ListMap

object MovieData {

  def main(args: Array[String]) {
    val movie_data = Util.getMovieData()
    println(movie_data.first())

    val num_movies = movie_data.count()
    println("num_movies: " + num_movies)

    val movie_fields = movie_data.map(lines =>  lines.split("\\|"))
    val years = movie_fields.map( field => field(2)).map( x => Util.convertYear(x))
    val years_filtered = years.filter(x => (x != 1900) )
    val movie_ages = years_filtered.map(yr =>  (1998-yr) ).countByValue()
    val movie_ages_sorted = ListMap(movie_ages.toSeq.sortBy(_._1):_*)

    val values = movie_ages.values
    val bins = movie_ages.keys

    println("movie_fields: " + movie_fields)
    println("years: " + years)
    println("years_filtered: " + years_filtered)
    println("movie_ages: " + movie_ages_sorted)
    println("values: " + values)
    println("bins: " + bins)
  }


}